include/orc/async.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright 2024 Adobe
+// All Rights Reserved.
+//
+// NOTICE: Adobe permits you to use, modify, and distribute this file in accordance with the terms
+// of the Adobe license agreement accompanying it.
+
+#pragma once
+
+#include <functional>
+
+//======================================================================================================================
+
+namespace orc {
+
+//======================================================================================================================
+
+// Enqueue a task for (possibly asynchronous) execution. If the `parallel_processing` setting in the
+// ORC config file is true, the task will be enqueued for processing on a background thread pool.
+// Otherwise, the task will be executed immediately in the current thread.
+void do_work(std::function<void()>);
```

Comment:
We were passing `do_work` all over the code base as the scheduler for any ORC task we may want to do asynchronously. (With the `parallel_processing` setting set to `false`, the tasks are run immediately- that's how we're able to implement that switch.)

Given that `do_work` is the _only_ scheduler we were using, I have removed it as a callback (more on that in a separate comment) and put it into its own header that the sources can reference themselves. This has made our asynchronous story much simpler, in terms of how we, you know, do work.